### PR TITLE
Linker - interface support

### PR DESCRIPTION
**Pull request by danielstro**

### CODE REVIEW

File: packages/schema-extract/test/linker/interface.spec.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        export interface TypeB {
+            somethingElse: TypeA
+        };
+        `}, 'TypeB', fileName);
+
+        const expected: Schema<'object'> = {
+            $ref: interfaceId,
+            properties: {
+                somethingElse: {
+                    $ref: interfaceId,
```
Comment (nadavwix) [inline packages/schema-extract/test/linker/interface.spec.ts]: definedAt:"#TypeA"

File: packages/schema-extract/test/linker/interface.spec.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        export interface TypeB extends TypeA<string> {
+            somethingElse: number
+        };
+        `}, 'TypeB', fileName);
+
+        const expected: Schema<'object'> = {
+            $ref: interfaceId,
+            properties: {
+                something: {
+                    type: 'string'
```
Comment (nadavwix) [inline packages/schema-extract/test/linker/interface.spec.ts]: add inheritedFrom: "#TypeA"

File: packages/schema-extract/test/linker/interface.spec.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        `}, 'TypeB', fileName);
+
+        const expected: Schema<'object'> = {
+            $ref: interfaceId,
+            properties: {
+                somethingElse: {
+                    type: 'number'
+                },
+                something: {
+                    enum: ['b'],
```
Comment (nadavwix) [inline packages/schema-extract/test/linker/interface.spec.ts]: add inheritedFrom: "TypeA"

File: packages/schema-extract/src/file-linker.ts [hunk 4]
```diff
@@ -199,6 +195,51 @@ export class SchemaLinker {
...
 
+    private linkInterface(entity: InterfaceSchema, schema: ModuleSchema): InterfaceSchema {
+        if (!schema.definitions) {
+            return entity;
+        }
+        const res = this.handleObject(entity, schema) as InterfaceSchema;
+        res.$ref = interfaceId;
+        if (entity.extends) {
+            const extendedEntity = entity.extends.$ref!.replace('#', '');
+            const refEntity: Schema = schema.definitions[extendedEntity];
```
Comment (danielstro) [inline packages/schema-extract/src/file-linker.ts]: This part will probably not work for imported interfaces

**Pull request merged**